refactor!: cleaning up tagging oracles

benesjan opened the pull request:
In this PR I merge `getIndexedTaggingSecretAsSender` and `incrementAppTaggingSecretIndexAsSender` and `get_app_tag_as_sender` into single `get_next_app_tag(sender, recipient)`. This results in the app tag being computed in TS and not in Noir.

The motivation for this is that to improve robustness of tagging as described in [this post](https://forum.aztec.network/t/on-note-discovery-and-index-coordination/7165) we need to associate a given index with tx hash and this is the first step in that direction.

### Note for reviewer

This is a breaking change because it changes oracles but I don't think a dev would ever call these oracles directly so I didn't bother with writing migration notes. If the reviewer thinks it should be done then LMK and I can add it.

benesjan commented:
* **#17415** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17415?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17372** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17372?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17372?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

nventuro commented on noir-projects/aztec-nr/aztec/src/oracle/notes.nr:
```diff
@@ -191,43 +189,19 @@ unconstrained fn check_nullifier_exists_oracle(_inner_nullifier: Field) -> bool
-}
-
-unconstrained fn increment_app_tagging_secret_index_as_sender_wrapper(
+/// Returns the next app tag for a given sender and recipient pair.
```
```suggestion
/// Returns the next app tag for a given sender and recipient pair.
///
/// This also notifies the simulator that a tag has been used in a note, and to therefore increment the
/// associated index so that future notes get a different tag and can be discovered by the recipient.
/// This change should only be persisted in a non-volatile database if the tagged log is found in an actual block -
/// otherwise e.g. a reverting transaction can cause the sender to accidentally skip indices and later produce notes
/// that are not found by the recipient.
```

Restoring the comment from 'increment'.

nventuro commented on noir-projects/noir-protocol-circuits/crates/types/src/indexed_tagging_secret.nr:
```diff
@@ -1,19 +0,0 @@
-pub global INDEXED_TAGGING_SECRET_LENGTH: u32 = 2;
-
-#[derive(Serialize, Deserialize)]
-pub struct IndexedTaggingSecret {
-    app_tagging_secret: Field,
-    index: u32,
-}
-
```
Why was this a proto struct?

benesjan replied:
It was introduced in [this PR](https://github.com/AztecProtocol/aztec-packages/pull/9468) by Grego. I don't really know but I guess it was because we have the TS counterpart and we wanted to have a matching struct in Noir.

nventuro commented on yarn-project/txe/src/rpc_translator.ts:
```diff
@@ -593,15 +593,6 @@ export class RPCTranslator {
     return toForeignCallResult(witness.toNoirRepresentation());
   }
 
-  async utilityGetIndexedTaggingSecretAsSender(foreignSender: ForeignCallSingle, foreignRecipient: ForeignCallSingle) {
```
Huh, rpc translator does not handle `privateIncrementAppTaggingSecretIndexAsSender`, so you were not forced to update to `privateGetNextAppTagAsSender`. I thought the class had been setup with zod to typecheck that all are handled, but it seems I was wrong.

Could you add the oracle, and maybe take a quick look to see if you can get the zod thing working? i.e. that when a txe session dispatches to the rpc translator we know that a) the fn is one of the allowed ones, and b) that all allowed fns are handled

benesjan replied:
Added the missing method in https://github.com/AztecProtocol/aztec-packages/pull/17372/commits/171b7bac5ebf06c2592bf7d27b8744cbc4b94e4a and will try to figure out the zod thing in a separate PR.

benesjan committed:
refactor!: cleaning up tagging oracles

In this PR I merge `getIndexedTaggingSecretAsSender` and `incrementAppTaggingSecretIndexAsSender` and `get_app_tag_as_sender` into single `get_next_app_tag(sender, recipient)`. This results in the app tag being computed in TS and not in Noir.

The motivation for this is that to improve robustness of tagging as described in [this post](https://forum.aztec.network/t/on-note-discovery-and-index-coordination/7165) we need to associate a given index with tx hash and this is the first step in that direction.

### Note for reviewer

This is a breaking change because it changes oracles but I don't think a dev would ever call these oracles directly so I didn't bother with writing migration notes. If the reviewer thinks it should be done then LMK and I can add it.

Co-authored-by: Jan Beneš <janbenes1234@gmail.com>
